version: '3.7'

services:
  postgres:
    image: postgis-pgr
    container_name: postgres
    environment:
      POSTGRES_DB: geodata
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: password
    ports:
      - "5432:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 15s
      timeout: 5s
      retries: 3

  load_data:
    image: python:3.9-slim
    container_name: load_data
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - .:/usr/src/app
    working_dir: /usr/src/app
    command: >
      sh -c "pip install -r requirements.txt && python load_geojson.py"

  load_calles:
    image: python:3.9-slim
    container_name: load_calles
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - .:/usr/src/app
    working_dir: /usr/src/app
    command: >
      sh -c "pip install -r requirements.txt && python load_calles.py"

  api:
    image: python:3.9-slim
    container_name: api
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - .:/usr/src/app
    working_dir: /usr/src/app
    environment:
      - FLASK_APP=prueba.py
    ports:
      - "5001:5001"
    command: >
      sh -c "pip install -r requirements.txt && flask run --host=0.0.0.0 --port=5001"

volumes:
  pgdata:
    driver: local
